src/server/plugins/reactHtml/createRequestHtmlFragment.jsx

```diff
@@ -113,6 +117,10 @@ const createRequestHtmlFragment = async (request, reply, { createRoutes }) => {
         });
 
         if (redirect) {
+          if (!isRedirectUrlAllowed(redirect.url)) {
+            renderStaticErrorPage(request, reply);
+            throw new Error(`'${redirect.url}' is not an allowed redirect URL`);
+          }
```

Comment:
i think if an error is thrown it should always result in `renderStaticErrorPage` if not the case thats probably how it should work

Reply:
Throwing an error does not render the error page for us, verified this locally.

Reply:
after looking into this further, this is the expected pattern.